### PR TITLE
Obtain PulseAudio cookie and socket from pactl

### PR DESCRIPTION
**Pull request by mad-s**
The pulseaudio server socket location changes from distribution to distribution and there is no default location, so with this patch it is parsed from pactl. The cookie has a default location (~/.config/pulse/cookie), so we should use that.

### CODE REVIEW

File: logic/subuserlib/classes/subuserSubmodules/run/runtime.py [hunk 1]
```diff
@@ -127,12 +128,12 @@ def getPulseAudioArgs(self):
     if "PULSE_SERVER" in self.env:
```
[the remark this answers is not in the document]
Comment (timthelion) [reply logic/subuserlib/classes/subuserSubmodules/run/runtime.py]: The TODO is in the line above. Github is tripping overitself trying to be smart, and therefore won't let me actually comment on the line with the TODO in it.

File: logic/subuserlib/classes/subuserSubmodules/run/runtime.py [hunk 2]
```diff
@@ -127,12 +128,12 @@ def getPulseAudioArgs(self):
...
     else:
-      pulseSocket = os.path.join("/run","user",str(self.user.endUser.uid),"pulse","native")
+      pulseSocket = next(filter(None, map(lambda x: x[20:].decode('ascii') if x.startswith(b'Server String: unix:') else None, subprocess.check_output(["pactl", "info"]).split(b'\n'))), None)
     if "PULSE_COOKIE" in self.env:
       pulseCookieFile = self.env["PULSE_COOKIE"]
     else:
-      pulseCookieFile = os.path.join(self.subuser.homeDirOnHost,".pulse","cookie")
-    if os.path.exists(pulseSocket) and os.path.exists(pulseCookieFile):
+      pulseCookieFile = next((f for f in [os.path.join(os.getenv("HOME"), ".config", "pulse", "cookie"), os.path.join(os.getenv("HOME"),".pulse","cookie")] if os.path.exists(f)), None)
+    if pulseSocket is not None and pulseCookieFile is not None:
```
Comment (timthelion) [inline logic/subuserlib/classes/subuserSubmodules/run/runtime.py]: Similarly to the comment above, getting the `$HOME` variable from the environment isn't reliable and `self.subuser.homeDirOnHost` is prefered.

**Comment by timthelion** [comment]
Thanks alot for the PR. I'm not an expert in the area of pulse audio so help in this area is extra-appreciated!

File: logic/subuserlib/classes/subuserSubmodules/run/runtime.py (hunk 2 above)
Comment (mad-s) [reply logic/subuserlib/classes/subuserSubmodules/run/runtime.py]: But `self.subuser.homeDirOnHost` is `$HOME/.subuser/homes/<subuser-name>/`, where we really want `$HOME`. I guess that should be `self.subuser.user.endUser.homeDir` instead?

Comment (timthelion) [reply logic/subuserlib/classes/subuserSubmodules/run/runtime.py]: Oops! my bad! You're right.

**Comment by timthelion** [comment]
Everything looks good with this PR. I'm sorry I haven't merged it yet. 
I'm going to merge it while preparing the next release, but the next 
release isn't ready yet. It is more comfortable for me to merge at that 
time because it requires fewer mental context switches and because it 
allows me to run some of the more in depth test suits (which take hours) 
fewer times. While this PR is very welcome, it doesn't fix any critical 
bugs so I'm not going to push a new release just for this one change.

Thank you for your time.

Tim

**Comment by timthelion** [comment]
Preliminary testing says this works!

**Pull request merged**